toolchain/lex/tokenized_buffer.cpp

```diff
@@ -49,6 +49,21 @@ auto VariantMatch(V&& v, Fs&&... fs) -> decltype(auto) {
   return std::visit(Overload{std::forward<Fs&&>(fs)...}, std::forward<V&&>(v));
 }
 
+#if __x86_64__
+#define CARBON_USE_SIMD 1
+
+// A table of masks to include 0-16 bytes of an SSE register.
+static const std::array<__m128i, sizeof(__m128i) + 1> prefix_masks = [] {
+  std::array<__m128i, sizeof(__m128i) + 1> masks = {};
+  for (auto [i, mask] : llvm::enumerate(masks)) {
+    memset(&mask, 0xFF, i);
+  }
+  return masks;
+}();
```

Comment:
Can this be made `constexpr`? A quick experiment on godbolt.org suggests we'd get a dynamic initializer here. The `memset` seems problematic but I think this would work:

```suggestion
// A table of masks to include 0-16 bytes of an SSE register.
static constexpr std::array<__m128i, sizeof(__m128i) + 1> prefix_masks = [] {
  std::array<__m128i, sizeof(__m128i) + 1> masks = {};
  for (auto [i, mask] : llvm::enumerate(masks)) {
    if (i > 0) {
      mask = (__m128i)((static_cast<unsigned __int128>(0) - 1) >> ((16 - i) * 8));
    }
  }
  return masks;
}();
```

or

```suggestion
static constexpr std::array<__m128i, sizeof(__m128i) + 1> prefix_masks = [] {
  std::array<__m128i, sizeof(__m128i) + 1> masks = {};
  for (auto [i, mask] : llvm::enumerate(masks)) {
    uint8_t byte_masks[16] = {};
    for (auto [j, byte_mask] : llvm::enumerate(byte_masks)) {
        byte_mask = j < i ? -1 : 0;
    }
    mask = (__m128i)__builtin_bit_cast(unsigned __int128, byte_masks);
  }
  return masks;
}();
```

(We can't `bit_cast` directly to `__m128i` at the moment, sadly, because Clang doesn't support vectors in `std::bit_cast` yet. But we can go via `__int128`.)

Reply:
Could I do this in a follow-up?

I've commoned this code between ARM and x86 and I'd like to experiment and come up with an approach that works for both.

Reply:
Sure, that's fine by me.

Reply:
K, will do. Left a TODO.